src/Testcontainers.AspireDashboard/AspireDashboardBuilder.cs

```diff
@@ -0,0 +1,91 @@
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+[PublicAPI]
+public sealed class AspireDashboardBuilder : ContainerBuilder<AspireDashboardBuilder, AspireDashboardContainer, AspireDashboardConfiguration>
+{
+    public const string AspireDashboardImage = "mcr.microsoft.com/dotnet/aspire-dashboard:latest";
+
+    public const ushort AspireDashboardFrontendPort = 18888;
+
+    public const ushort AspireDashboardOtlpPort = 18889;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AspireDashboardBuilder" /> class.
+    /// </summary>
+    public AspireDashboardBuilder()
+        : this(new AspireDashboardConfiguration(false, false))
+    {
+        DockerResourceConfiguration = Init().DockerResourceConfiguration;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AspireDashboardBuilder" /> class.
+    /// </summary>
+    /// <param name="resourceConfiguration">The Docker resource configuration.</param>
+    private AspireDashboardBuilder(AspireDashboardConfiguration resourceConfiguration)
+        : base(resourceConfiguration)
+    {
+        DockerResourceConfiguration = resourceConfiguration;
+    }
+
+    /// <inheritdoc />
+    protected override AspireDashboardConfiguration DockerResourceConfiguration { get; }
+
+    /// <summary>
+    /// Configures the dashboard to accept anonymous access.
+    /// </summary>
+    /// <param name="allowed">A value indicating whether anonymous access is allowed.</param>
+    /// <returns>A configured instance of <see cref="AspireDashboardBuilder" />.</returns>
+    public AspireDashboardBuilder AllowAnonymous(bool allowed)
+    {
+        return Merge(DockerResourceConfiguration, new AspireDashboardConfiguration(allowAnonymous: allowed))
+            .WithEnvironment("DOTNET_DASHBOARD_UNSECURED_ALLOW_ANONYMOUS", allowed.ToString().ToLowerInvariant());
+    }
+
+    /// <summary>
+    /// Configures the dashboard to allow unsecured transport.
+    /// </summary>
+    /// <param name="allowed">A value indicating whether unsecured transport is allowed.</param>
+    /// <returns>A configured instance of <see cref="AspireDashboardBuilder" />.</returns>
+    public AspireDashboardBuilder AllowUnsecuredTransport(bool allowed)
+    {
+        return Merge(DockerResourceConfiguration, new AspireDashboardConfiguration(allowUnsecuredTransport: allowed))
+            .WithEnvironment("ASPIRE_ALLOW_UNSECURED_TRANSPORT", allowed.ToString().ToLowerInvariant());
+    }
+
+    /// <inheritdoc />
+    public override AspireDashboardContainer Build()
+    {
+        Validate();
+        return new AspireDashboardContainer(DockerResourceConfiguration);
+    }
+
+    /// <inheritdoc />
+    protected override AspireDashboardBuilder Init()
+    {
+        return base.Init()
+            .WithImage(AspireDashboardImage)
+            .WithPortBinding(AspireDashboardFrontendPort, AspireDashboardFrontendPort)
```

Comment:
We never bind ports to fixed values because we do not know anything about the developers' (test) environments. This does not follow our [best practices](https://dotnet.testcontainers.org/api/best_practices/). Please use a random host port instead. For modules that contain a web frontend, we usually log the address, like [Couchbase](https://github.com/testcontainers/testcontainers-dotnet/blob/a0f1f7694b4602aa2ba7da6f167ddc4a56670ecc/src/Testcontainers.Couchbase/CouchbaseContainer.cs#L18) does.

Regarding the design-related questions:

> I have design-related questions:
> 
> 1. It only makes sense to use Aspire with reusable value and persist it during test runs, do we want to make it as default?
> 2. It makes sense to expose Aspire on the well-known port by default 18888. What do you think?

I think we should add a [module-specific documentation](https://github.com/testcontainers/testcontainers-dotnet/blob/a0f1f7694b4602aa2ba7da6f167ddc4a56670ecc/mkdocs.yml#L41-L49) that explains how the different configurations can be used to integrate the Aspire dashboard.

Reply:
I agree, I've added the documentation